DSharpPlus.Core/Rest/RestClient.cs

```diff
@@ -0,0 +1,50 @@
+// This file is part of the DSharpPlus project.
+//
+// Copyright (c) 2015 Mike Santiago
+// Copyright (c) 2016-2022 DSharpPlus Contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System.Net.Http;
+using System.Threading.Tasks;
+
+using Polly;
+
+namespace DSharpPlus.Core.Rest
+{
+    // another stub, just like DiscordClient.
+    public class RestClient
+    {
+        private readonly HttpClient client;
+
+        public RestClient(HttpClient client) 
+            => this.client = client;
+
+        public async Task<HttpResponseMessage> MakeRequestAsync(IRestRequest request)
+        {
+            HttpRequestMessage message = request.Build();
+
+            // this way our execution policy knows what we're doing
+            message.SetPolicyExecutionContext(request.Context);
+
+            // we should be handling errors etc here, but again, this is a stub to show the concept rather than a final implementation.
+            return await client.SendAsync(message);
```

Comment:
This *could* be made a passthrough method, but it'll screw up stack-traces, so maybe not.

Reply:
shows the concept rather than proposing an implementation